crates/sparrow-expressions/src/evaluators.rs

```diff
@@ -51,7 +51,8 @@ struct EvaluatorFactory {
 inventory::collect!(EvaluatorFactory);
 
 pub struct StaticInfo<'a> {
-    input_schema: &'a Schema,
+    /// Type of the input that the expressions are run on.
+    input_type: &'a DataType,
```

Comment:
Is this similar to the existing `StaticInfo`? 

Would saying `Type of the input that this evaluator accepts` be more accurate?

Reply:
I believe so. Agreed the comment/naming isn't great, but I think the suggestion may be more confusing. Specifically:

The weird part is that this isn't "the input types to the evaluator" (those are the arguments to the expression / evaluator). This is the input that the entire set of expressions are being run on.

Consider: `project`

1. This runs on the input batch, so the input type is the `DataType::Struct(...)` describing the input.
2. The `input` expression reads that and produces the `DataType::Struct(...)` corresponding to the input.
3. The `field_ref` expression takes the result of (2) and returns a given field.
4. Another `field_ref` expression takes the result of (2) and returns a different field.
5. An `add` expression takes the results of (3) and (4) and returns the result.

So, the `input_type` in all of their `StaticInfo` is information about "how these expressions are executed" (on the input batch) not the individual evaluator.

Reply:
Updated this struct to:

```
/// Static information available when creating an evaluator.
pub struct StaticInfo<'a> {
    /// Type of the input that all the expressions are run on.
    input_type: &'a DataType,
    /// Name of the instruction to be evaluated.
    name: &'a Cow<'static, str>,
    /// Literal (static) arguments to *this* expression.
    literal_args: &'a [ScalarValue],
    /// Arguments (dynamic) to *this* expression.
    args: Vec<&'a StaticArg<'a>>,
    /// Result type this expression should produce.
    ///
    /// For many instructions, this should be inferred from the arguments.
    /// It is part of the plan (a) for simplicity, so a plan may be executed
    /// without performing type-checking and (b) because some instructions
    /// need to know the result-type in order to execute (eg., cast).
    result_type: &'a DataType,
}
```